lib/service.dart

```diff
@@ -192,6 +232,243 @@ class IsolateManager {
 
     return isolates.isEmpty ? null : isolates.first;
   }
+
+  void _initFlutterIsolate(List<IsolateRef> isolates) async {
+    for (IsolateRef ref in isolates) {
+      // Populate flutter isolate info.
+      if (_flutterIsolate == null) {
+        final Isolate isolate = await _service.getIsolate(ref.id);
+        if (isolate.extensionRPCs != null) {
+          for (String extensionName in isolate.extensionRPCs) {
+            if (extensionName.startsWith('ext.flutter.')) {
+              _setFlutterIsolate(ref);
+              break;
+            }
+          }
+        }
+      }
+      _serviceExtensionManager._addRegisteredExtensionRPCs(ref);
+    }
+  }
+
+  void _setFlutterIsolate(IsolateRef ref) {
+    if (_flutterIsolate == ref) {
+      // Isolate didn't change. Do nothing.
+      return;
+    }
+    _flutterIsolate = ref;
+    _flutterIsolateController.add(ref);
+    _flutterIsolateController.onListen =
```

Comment:
set the onListen handler back when the _flutterIsolateController is defined not here.
We want an initial value of null in the stream if you try to listen early.

Reply:
Initialized _flutterIsolateController in a constructor so that we could access _flutterIsolate and add it as a stream event.